### PR TITLE
fix(jsonSchema): class shortname

### PR DESCRIPTION
**Pull request by Zowac**
| Q             | A
| ------------- | ---
| Branch?       | 3.1
| License       | MIT

See https://github.com/api-platform/core/pull/5455#issuecomment-1469699042

### CODE REVIEW

File: src/Metadata/Resource/ResourceMetadataCollection.php [hunk 1]
```diff
@@ -102,7 +102,7 @@ public function getOperation(?string $operationName = null, bool $forceCollectio
     private function handleNotFound(string $operationName, ?ApiResource $metadata): void
     {
         // Hide the FQDN in the exception message if possible
-        $shortName = $metadata ? $metadata->getShortName() : $this->resourceClass;
+        $shortName = $metadata?->getShortName() ? $metadata->getShortName() : $this->resourceClass;
```
Comment (soyuka) [inline src/Metadata/Resource/ResourceMetadataCollection.php:105]: Why this change?

**Comment by soyuka** [comment]
Can you squash your commits and rename:

```
fix(jsonschema): find the related operation instead of assuming one
```

File: src/Metadata/Resource/ResourceMetadataCollection.php (hunk 1 above)
Comment (Zowac) [reply src/Metadata/Resource/ResourceMetadataCollection.php:105]: `getShortName` can return null, if we don't check the method it happens that we don't have the class that is impacted by the error because the shortName is empty

**Comment by soyuka** [comment]
Thanks!

**Comment by LoicBoursin** [comment]
Thanks @Zowac & @soyuka !! 😍

**Comment by jamesisaac** [comment]
Hi,

This PR seems to have caused a regression in some cases of JsonSchema generation.  I've created a minimal repro here: #5626
